### PR TITLE
Feature/filter peer by client

### PR DESCRIPTION
**Pull request by asdacap**
- Add a config to select peer to connect to by client ID.
- Useful to test connecting to specific client type.

## Changes

- Add a config to select peer to connect to by client ID.

#### What types of changes does your code introduce?

- [X] New feature (a non-breaking change that adds functionality)

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

- Can filter
- If filter not set, does not do anything.

#### Requires documentation update

- [X] No

#### Requires explanation in Release Notes

- [X] No

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

File: src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs [hunk 1]
```diff
@@ -61,9 +64,15 @@ public class P2PProtocolHandler : ProtocolHandlerBase, IPingSender, IP2PProtocol
         PublicKey localNodeId,
         INodeStatsManager nodeStatsManager,
         IMessageSerializationService serializer,
+        INetworkConfig networkConfig,
         ILogManager logManager) : base(session, nodeStatsManager, serializer, logManager)
     {
         _nodeStatsManager = nodeStatsManager ?? throw new ArgumentNullException(nameof(nodeStatsManager));
+        if (networkConfig.ClientIdMatcher != null)
+        {
+            _clientIdPattern = new Regex(networkConfig.ClientIdMatcher);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs:73]: should we use `Compiled` ?

Comment (asdacap) [reply src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs:73]: Not sure. Maybe?

Comment (asdacap) [reply src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs]: Done

File: src/Nethermind/Nethermind.Network/Config/INetworkConfig.cs [hunk 2]
```diff
@@ -94,5 +94,8 @@ public interface INetworkConfig : IConfig
 
         [ConfigItem(DefaultValue = "1", HiddenFromDocs = true, Description = "[TECHNICAL] Num of thread in final processing of network packet. Set to more than 1 if you have very fast internet.")]
         int ProcessingThreadCount { get; set; }
+
+        [ConfigItem(DefaultValue = null, HiddenFromDocs = true, Description = "[TECHNICAL] Only allow peer with clientId matching this regex. Useful for testing.")]
```
[the remark this answers is not in the document]
Comment (asdacap) [reply src/Nethermind/Nethermind.Network/Config/INetworkConfig.cs]: Done

**Pull request merged**